inst/htmlwidgets/leaflet.js

```diff
@@ -401,19 +402,32 @@ var dataframe = (function() {
 
     if (icon) icondf.effectiveLength = df.nrow();
 
+    var clusterGroup, cluster = clusterOptions !== null;
+    if (cluster) {
+      clusterGroup = L.markerClusterGroup(clusterOptions);
```

Comment:
Actually I'm a little unclear on what the ownership of a marker should be.

```r
shinyServer(function(input, output, session) {
  output$map <- renderLeaflet({
    leaflet() %>% addMarker(0, 0, layerId = "marker1")
  })
  observe({
    clusterOptions <- if (input$cluster) list() else NULL
    leafletProxy() %>% addMarker(0, 0, layerId = "marker1", clusterGroup = clusterOptions)
  })
})
```

I'd expect each time the observe executes, it would replace the marker, but I don't think this is going to happen...?